src/test/java/org/broadinstitute/hellbender/engine/ReadShardUnitTest.java

```diff
@@ -0,0 +1,110 @@
+                                      new ReadShard(new SimpleInterval("1", 101, 150), new SimpleInterval("1", 101, 150), readsSource),
+                                      new ReadShard(new SimpleInterval("1", 151, 200), new SimpleInterval("1", 151, 200), readsSource))
+                },
+
+                {
+                        new SimpleInterval("1", 51, 200), 50, 10, readsSource, dictionary,
+                        Arrays.asList(new ReadShard(new SimpleInterval("1", 51, 100), new SimpleInterval("1", 41, 110), readsSource),
+                                      new ReadShard(new SimpleInterval("1", 101, 150), new SimpleInterval("1", 91, 160), readsSource),
+                                      new ReadShard(new SimpleInterval("1", 151, 200), new SimpleInterval("1", 141, 210), readsSource))
+                },
+
+                {
+                        new SimpleInterval("1", 51, 200), 70, 0, readsSource, dictionary,
+                        Arrays.asList(new ReadShard(new SimpleInterval("1", 51, 120), new SimpleInterval("1", 51, 120), readsSource),
+                                      new ReadShard(new SimpleInterval("1", 121, 190), new SimpleInterval("1", 121, 190), readsSource),
+                                      new ReadShard(new SimpleInterval("1", 191, 200), new SimpleInterval("1", 191, 200), readsSource))
+                },
+
+                {
+                        new SimpleInterval("1", 51, 200), 70, 10, readsSource, dictionary,
+                        Arrays.asList(new ReadShard(new SimpleInterval("1", 51, 120), new SimpleInterval("1", 41, 130), readsSource),
+                                      new ReadShard(new SimpleInterval("1", 121, 190), new SimpleInterval("1", 111, 200), readsSource),
+                                      new ReadShard(new SimpleInterval("1", 191, 200), new SimpleInterval("1", 181, 210), readsSource))
+                },
+
+                {
+                        new SimpleInterval("1", 15999, 16000), 100, 10, readsSource, dictionary,
+                        Arrays.asList(new ReadShard(new SimpleInterval("1", 15999, 16000), new SimpleInterval("1", 15989, 16000), readsSource))
+                },
+        };
+    }
+
+    @Test(dataProvider = "ShardIntervalTestData")
+    public void testShardInterval( final SimpleInterval originalInterval, final int windowSize, final int windowPadding, final ReadsDataSource readsSource, final SAMSequenceDictionary dictionary, List<ReadShard> expectedWindows  ) {
+        final List<ReadShard> windows = ReadShard.shardInterval(originalInterval, windowSize, windowPadding, readsSource, dictionary);
+
+        Assert.assertEquals(windows.size(), expectedWindows.size(), "Wrong number of windows created by ReadWindowWalker.shardInterval()");
+        for ( int i = 0; i < windows.size(); ++i ) {
+            final ReadShard window = windows.get(i);
```

Comment:
Done -- implemented as a separate test. Also improved test coverage for `ReadShard` in general.